conans/model/conan_file.py

```diff
@@ -258,39 +259,79 @@ def new_cpp_info(self):
     def source_folder(self):
         return self.folders.source_folder
 
+    @property
+    def source_path(self) -> Path:
+        assert self.source_folder is not None, "`source_folder` is `None`"
+        return Path(self.source_folder)
+
     @property
     def export_sources_folder(self):
         """points to the base source folder when calling source() and to the cache export sources
         folder while calling the exports_sources() method. Prepared in case we want to introduce a
         'no_copy_export_sources' and point to the right location always."""
         return self.folders.base_export_sources
 
+    @property
+    def export_sources_path(self) -> Path:
+        assert self.export_sources_folder is not None, "`export_sources_folder` is `None`"
+        return Path(self.export_sources_folder)
+
     @property
     def export_folder(self):
         return self.folders.base_export
 
+    @property
+    def export_path(self) -> Path:
+        assert self.export_folder is not None, "`export_folder` is `None`"
+        return Path(self.export_folder)
+
     @property
     def build_folder(self):
         return self.folders.build_folder
 
+    @property
+    def build_path(self) -> Path:
+        assert self.build_folder is not None, "`build_folder` is `None`"
+        return Path(self.build_folder)
+
     @property
     def package_folder(self):
         return self.folders.base_package
 
+    @property
+    def package_path(self) -> Path:
+        assert self.package_folder is not None, "`package_folder` is `None`"
+        return Path(self.package_folder)
+
     @property
     def install_folder(self):
         # FIXME: Remove in 2.0, no self.install_folder
         return self.folders.base_install
 
+    @property
+    def install_path(self) -> Path:
+        assert self.install_folder is not None, "`install_folder` is `None`"
+        return Path(self.install_folder)
+
     @property
     def generators_folder(self):
         # FIXME: Remove in 2.0, no self.install_folder
         return self.folders.generators_folder if self.folders.generators else self.install_folder
 
+    @property
+    def generators_path(self) -> Path:
+        assert self.generators_folder is not None, "`generators_folder` is `None`"
+        return Path(self.generators_folder)
+
     @property
     def imports_folder(self):
         return self.folders.imports_folder
 
+    @property
+    def imports_path(self) -> Path:
```

Comment:
This one disappears in 2.0, better remove it.

Reply:
removed